rabbitmq/tests/test_unit.py

```diff
@@ -173,3 +180,44 @@ def test_disable_nodes(aggregator, check):
     # check to ensure other metrics are being collected
     for m in metrics.Q_METRICS:
         aggregator.assert_metric(m, count=1)
+
+
+def test_queues_regexes_exclude_with_negative_lookahead(aggregator, dd_run_check):
+    """Based on a support case where a customer was confused why their regular expression didn't work.
+
+    The key piece optionally matching the vhost part.
+    """
+    data = {}
+    for ep in ("queues", "overview"):
+        with open("tests/fixtures/mgmt/{}.json".format(ep)) as fh:
+            data[common.URL + ep] = json.load(fh)
+
+    def mock_get_data(_self, url):
+        return data.get(url, [])
+
+    instance = {
+        "rabbitmq_api_url": common.URL,
+        "rabbitmq_user": "guest",
+        "rabbitmq_pass": "guest",
+        "queues_regexes": [r"""(?x) # Enable verbose flag to split expression into commented parts.
+        ^ # We have to anchor at beginning of string to enforce checking for the prefix.
+        (?!
+        (?://)? # Match vhost part if it's present.
+        config/foo\.updated-configs\.) # Prefix we want to exclude.
+        .+ # Match everything else as long as it's NOT preceded by prefix.
+        """],
+    }
+    check = RabbitMQ("rabbitmq", {}, instances=[instance])
+    with mock.patch.object(RabbitMQManagement, "_get_data", new_callable=lambda: mock_get_data):
+        dd_run_check(check)
+
+    print(list(aggregator._metrics))
+    for m in metrics.Q_METRICS:
+        # Make sure we did collect the metric.
+        aggregator.assert_metric(m, at_least=1)
+        # Make sure we didn't collect it for the excluded queue.
+        aggregator.assert_metric_has_tag(
+            m,
+            "rabbitmq_queue:config/foo.updated-configs.2023-01-18",
+            count=0,
+        )
```

Comment:
What do you think of also validating the metrics with the metadata? (using `aggregator.assert_metrics_using_metadata(get_metadata_metrics())`)

Reply:
I'll give it a shot, but I'm not sure my fixture actually *contains* all the metrics...

I'll be looking at the legacy tests in general, I discovered that at least 1 unit test talks to the rabbitmq server we spin up for the integration tests 🙈 
I think I'll find a test where we can add these assertions and it makes sense.

Reply:
it does not need to contain all the metrics, it will only validate the metrics that are collected 🙂

Reply:
easy-peasy!